### PR TITLE
add a new RandomGunGenerator

### PR DESCRIPTION
**Pull request by cbotta**
This PR is to add also in 75X a new RandomGunGenarator (already added in 71X)

### CODE REVIEW

**Comment by bendavid** [comment]
+1

File: IOMC/ParticleGuns/src/ExpoRandomPGunProducer.cc [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+   if ( fVerbosity > 0 )
+   {
+      fEvt->print() ;
+   }
+
+   auto_ptr<HepMCProduct> BProduct(new HepMCProduct()) ;
+   BProduct->addHepMCData( fEvt );
+   e.put(BProduct);
+
+   auto_ptr<GenEventInfoProduct> genEventInfo(new GenEventInfoProduct(fEvt));
```
Comment (davidlange6) [inline IOMC/ParticleGuns/src/ExpoRandomPGunProducer.cc]: should fEvt be deleted?

Comment (bendavid) [inline IOMC/ParticleGuns/src/ExpoRandomPGunProducer.cc]: No, the HepMCProduct takes ownership of fEvt it at line 146

**Comment by davidlange6** [comment]
+1

**Pull request merged**